apps/people-app/backend/modules/database/db_functions.bal

```diff
@@ -73,3 +74,61 @@ public isolated function updateVehicle(UpdateVehiclePayload payload) returns boo
     }
     return false;
 }
+
+# Function to fetch a recruit by ID.
+#
+# + recruitId - ID of the recruit to fetch
+# + return - Recruit or an error or null if not found
+public isolated function fetchRecruitById(int recruitId) returns Recruit|error? {
+    Recruit|error result = databaseClient->queryRow(getRecruitByIdQuery(recruitId));
+    if result is sql:NoRowsError {
+        return;
+    }
+
+    return result;
+}
+
+# Function to fetch all recruits.
+#
+# + return - Array of recruits or an error
+public isolated function fetchRecruits() returns Recruit[]|error? {
+    stream<Recruit, error?> recruitsResponse = databaseClient->query(getRecruits());
+    return from Recruit recruit in recruitsResponse
+        select recruit;
+}
+
+# Persist new recruit in DB.
+#
+# + recruit - Recruit payload
+# + createdBy - User who creates the recruit entry
+# + return - Id of the newly created recruit or an error
+public isolated function addRecruit(AddRecruitPayload recruit, string createdBy) returns int|error {
+    sql:ExecutionResult executionResult = check databaseClient->execute(addRecruitQuery(recruit, createdBy));
+    return executionResult.lastInsertId.ensureType(int);
+}
+
+# Update recruit info dynamically based on changed fields.
+#
+# + id - Recruit id
+# + recruit - Recruit payload with updated fields
+# + return - error or null
+public isolated function UpdateRecruit(int id, UpdateRecruitPayload recruit) returns error? {
+    if recruit.entries().length() == 0 {
+        return error(string `No data to update for recruit with id: ${id}`);
+    }
+    sql:ExecutionResult result = check databaseClient->execute(updateRecruitQuery(id, recruit));
+
+    if result.affectedRowCount == 0 {
+        return error(string `No recruit found to update for id: ${id}`);
+    }
+    return;
+}
```

Comment:
```suggestion
public isolated function UpdateRecruit(int id, UpdateRecruitPayload recruit) returns boolean|error {
    sql:ExecutionResult result = check databaseClient->execute(updateRecruitQuery(id, recruit));
    return result.affectedRowCount > 0;
}
```

In the `service.bal` we can check and return,

- `InternalServerError` if an error occurred while executing this query.
- `NotFound` if the result is false.
- `Ok` else.